internal/activities/validate_structure.go

```diff
@@ -171,3 +176,16 @@ func extraNodes(sipBase, path string, expected []string, matchDir bool) ([]strin
 
 	return extras, nil
 }
+
+func validateName(name string) bool {
+	const validChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_.()"
+
+	// Make sure only valid characters exist in name.
```

Comment:
This type of comments are better above the function name:

```
// validateName makes sure only valid characters exist in name.
func validateName(name string) bool {
```

Not a big deal in this case for a small private function, but that way it documents the function and you can read it when you use it.